### PR TITLE
Add setting to enable unlisted emotes

### PR DESCRIPTION
**Pull request by goldbattle**
Pull request checklist:

- [ ] `CHANGELOG.md` was updated, if applicable

# Description

This add support for hiding unlisted emotes and a setting to enable them if the user wishes to do so. This was requested as per https://github.com/Chatterino/chatterino2/pull/3071#issuecomment-1008454760 and is part of my effort to upstream 7tv support (see https://github.com/Chatterino/chatterino2/pull/3071#issuecomment-1008367948).

### CODE REVIEW

**Comment by MrAuro** [comment]
While testing this I noticed that you have to reload all emotes (F5) anytime you update the newly added setting for the changes to take effect

image

**Comment by goldbattle** [comment]
Good point, I guess that is something people would want. I simply call refresh on the 7TV emotes if the setting is changed.

**Comment by jammehcow** [comment]
I'm not the biggest fan of refreshing all emotes each time the setting is changed. Would prefer that it was handled locally, but since this setting isn't going to be changed frequently it's probably fine.

pajaBing 1x @zneix would like your opinion.

**Comment by goldbattle** [comment]
Are you able to take a look at this @zneix?

**Review by TroyKomodo — APPROVED** [review]

**Comment by TroyKomodo** [comment]
@goldbattle there seems to be some mistakes, please verify that this compiles and i will be happy to merge it.

File: src/widgets/settingspages/GeneralPage.cpp [hunk 1]
```diff
@@ -315,6 +317,23 @@ void GeneralPage::initLayout(GeneralPageView &layout)
 
     layout.addCheckbox("Remove spaces between emotes",
                        s.removeSpacesBetweenEmotes);
+    layout.addCheckbox("Show unlisted / unapproved emotes (7TV only)",
+                       s.showUnlistedEmotes);
+    s.showUnlistedEmotes.connect(
+        []() {
+            getApp()->twitch.server->forEachChannelAndSpecialChannels(
```
Comment (TroyKomodo) [inline src/widgets/settingspages/GeneralPage.cpp:324]:
```suggestion
            getApp()->twitch->server->forEachChannelAndSpecialChannels(
```

**Review by TroyKomodo — CHANGES_REQUESTED** [review]

File: src/widgets/settingspages/GeneralPage.cpp (hunk 1 above)
Comment (Felanbird) [inline src/widgets/settingspages/GeneralPage.cpp:324]:
```suggestion
            getApp()->twitch->server->forEachChannelAndSpecialChannels(
```

Comment (Felanbird) [reply src/widgets/settingspages/GeneralPage.cpp:324]: hey i was gonna do that

**Comment by goldbattle** [comment]
I can check this weekend, but feel free to fix and merge if it works for
you.

File: src/widgets/settingspages/GeneralPage.cpp (hunk 1 above)
Comment (TroyKomodo) [reply src/widgets/settingspages/GeneralPage.cpp]: it was wrong.

Comment (TroyKomodo) [reply src/widgets/settingspages/GeneralPage.cpp]: 💀

File: src/widgets/settingspages/GeneralPage.cpp [hunk 2]
```diff
@@ -315,6 +317,23 @@ void GeneralPage::initLayout(GeneralPageView &layout)
 
     layout.addCheckbox("Remove spaces between emotes",
                        s.removeSpacesBetweenEmotes);
+    layout.addCheckbox("Show unlisted / unapproved emotes (7TV only)",
+                       s.showUnlistedEmotes);
+    s.showUnlistedEmotes.connect(
+        []() {
+            getApp()->twitch->server->forEachChannelAndSpecialChannels(
```
Comment (TroyKomodo) [inline src/widgets/settingspages/GeneralPage.cpp:324]:
```suggestion
            getApp()->twitch->forEachChannelAndSpecialChannels(
```

File: src/widgets/settingspages/GeneralPage.cpp (hunk 1 above)
Comment (Felanbird) [reply src/widgets/settingspages/GeneralPage.cpp]: ya we suck

**Review by TroyKomodo — APPROVED** [review]

**Comment by TroyKomodo** [comment]
@goldbattle Feel free to tag me on discord if you do not have the contributor role.

**Pull request merged**